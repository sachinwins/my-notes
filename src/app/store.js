import { configureStore } from '@reduxjs/toolkit'
import LoginSlice from '../features/login/LoginSlice'
import NoteReducer from '../features/notes/reducers/index'

export const reducer = {
  login: LoginSlice,
  gnotes: NoteReducer,
}

export default configureStore({
  reducer,
  devTools: true,
})
