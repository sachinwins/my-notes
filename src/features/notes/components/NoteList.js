import React from 'react'
import { ListItemText } from '@material-ui/core'

const NoteList = ({ note, index }) => {
  return (
    <>
      <ListItemText primary={note.title} />
      <ListItemText secondary={note.content} />
    </>
  )
}
export default NoteList
