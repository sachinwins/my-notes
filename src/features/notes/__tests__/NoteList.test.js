import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import NoteList from '../components/NoteList'
import { Provider } from 'react-redux'
import store from '../../../app/store'

describe('NoteList Component ', () => {
  const props = {
    note: { title: 'Note 1', content: 'Content 1' },
    index: 0,
  }

  const { getAllByText } = render(
    <Provider store={store}>
      <NoteList {...props} />
    </Provider>
  )
  test('renders AddNote component', () => {
    expect(getAllByText('Note 1'))
    expect(getAllByText('Content 1'))
  })
  test('It should match with snapshot', () => {
    const tree = renderer.create(<NoteList {...props} />).toJSON()
    expect(NoteList).toMatchSnapshot()
  })
})
