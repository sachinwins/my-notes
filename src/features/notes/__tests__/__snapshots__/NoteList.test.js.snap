// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList Component  It should match with snapshot 1`] = `[Function]`;
